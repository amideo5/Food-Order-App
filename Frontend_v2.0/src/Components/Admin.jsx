import React from "react";
import "../assets/main.css";
import { useState } from "react";
import { useForm } from "react-hook-form";
import axios from "axios";
import NavBar from "./NavBar";

const Admin = (props) => {
  const { reset } = useForm();

  const [values, setValues] = useState({
    password: "",
    id: "",
  });

  const handleChange = (event) => {
    setValues({ ...values, [event.target.name]: event.target.value });
  };

  const handleSubmit = (event) => {
          if(values.id == "101" && values.password == "101"){
          localStorage.setItem("token", "saasssadasdasd");
          localStorage.setItem("id", values.id);
          props.history.push({ pathname: "/welcomea" });
          reset();
        }
        else{
            alert("Invalid Login")
            reset();
        }
        
     
      
    reset();
    event.preventDefault();
  };

  return (
    <div className="page">
      <NavBar />
      <div className="form-container">
        <div className="form-box">
          <div className="header-form">
            <h4 className="text-primary text-center">
              <i
                className="fa fa-user-circle"
                style={{ fontSize: "110px" }}
              ></i>
            </h4>
            <div className="image"></div>
          </div>
          <h2>Admin Sign In</h2>
          <div className="body-form">
            <form className="m-3" onSubmit={handleSubmit}>
              <div className="input-group mb-3">
                <div className="input-group-prepend">
                  <span className="input-group-text">
                    <i className="fa fa-envelope fa-2x"></i>
                  </span>
                </div>

                <input
                  type="text"
                  className="form-control"
                  placeholder="Admin ID"
                  onChange={handleChange}
                  name="id"
                  required
                  value={values.id}
                />
              </div>
              <div className="input-group mb-3">
                <div className="input-group-prepend">
                  <span className="input-group-text">
                    <i className="fa fa-lock fa-2x"></i>
                  </span>
                </div>
                <input
                  type="password"
                  className="form-control"
                  placeholder="Password"
                  onChange={handleChange}
                  name="password"
                  required
                  value={values.password}
                  
                />
              </div>
              <button type="submit" className="btn btn-secondary btn-block">
                Log In
              </button>

              <br />
              <br />

              
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Admin;
