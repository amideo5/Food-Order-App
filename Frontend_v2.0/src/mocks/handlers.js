// src/mocks/handlers.js
import { rest } from "msw";

export const handlers = [
  rest.post("http://localhost:8080/users/signin", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Successful")
    );
  }),

  rest.post("http://localhost:8080/users/createUser", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Registered")
    );
  }),

  rest.get("http://localhost:8080/users/name/*", async (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Aryan")
    );
  }),

  rest.get("http://localhost:8080/orders/getOrder/*", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json({
        orderName: "Aryan",
        items: {
          orderItems: {
            rajma: "2",
            chicken: "3",
          },
        },
        cost: "2000",
      })
    );
  }),

  rest.put("http://localhost:8080/users/updateUser/*", (req, res, ctx) => {
    return res(ctx.status(200), ctx.body("User Updated"));
  }),
  
];
