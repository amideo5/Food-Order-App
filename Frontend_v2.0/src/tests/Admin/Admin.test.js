import { render, screen } from "@testing-library/react";
import Admin from "../../Components/Admin";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";
import React from "react";
import { fireEvent, waitFor } from "@testing-library/react";
import { server } from "../../mocks/server";
import { rest } from "msw";

describe("Testing Admin Component", () => {
  test("Snapshot testing", () => {
    const { container } = render(<Admin />);

    expect(container).toMatchSnapshot();
  });

  test("Checking component is rendering or not", () => {
    render(<Admin />);
    const signingText = screen.getAllByText("Admin Sign In");
    const navText = screen.getAllByText("NEX Food Ordering");
    expect(signingText[0]).toBeInTheDocument();
    expect(navText[0]).toBeInTheDocument();
  });


  test('renders Admin form elements correctly', () => {
    render(<Admin />);
    const idInput = screen.getByPlaceholderText('Admin ID');
    const passwordInput = screen.getByPlaceholderText('Password');
    const loginButton = screen.getByRole('button', { name: 'Log In' });
    expect(idInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
  });

  test('updates state correctly when input values change', () => {
    render(<Admin />);
    const idInput = screen.getByPlaceholderText('Admin ID');
    const passwordInput = screen.getByPlaceholderText('Password');
    fireEvent.change(idInput, { target: { value: '101' } });
    fireEvent.change(passwordInput, { target: { value: '101' } });
    expect(idInput).toHaveValue('101');
    expect(passwordInput).toHaveValue('101');
  });

//   test('submitting form with valid login details redirects to welcomea page', () => {
//     const mockPush = jest.fn();
//     jest.mock('react-router-dom', () => ({
//       ...jest.requireActual('react-router-dom'),
//       useHistory: () => ({
//         push: mockPush,
//       }),
//     }));
//     render(<Admin />);
//     const idInput = screen.getByPlaceholderText('Admin ID');
//     const passwordInput = screen.getByPlaceholderText('Password');
//     const loginButton = screen.getByRole('button', { name: 'Log In' });
//     fireEvent.change(idInput, { target: { value: '101' } });
//     fireEvent.change(passwordInput, { target: { value: '101' } });
//     fireEvent.click(loginButton);
    
//   });

  test('submitting form with invalid login details shows error message', async() => {
    render(<Admin />);
    const idInput = screen.getByPlaceholderText('Admin ID');
    const passwordInput = screen.getByPlaceholderText('Password');
    const loginButton = screen.getByRole('button', { name: 'Log In' });
    fireEvent.change(idInput, { target: { value: '102' } });
    fireEvent.change(passwordInput, { target: { value: '102' } });
    fireEvent.click(loginButton);
    const mockAlert = jest.fn();
    
  });
  
});