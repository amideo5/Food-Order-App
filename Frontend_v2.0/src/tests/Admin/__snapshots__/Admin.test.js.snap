// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Admin Component Snapshot testing 1`] = `
<div>
  <div
    class="page"
  >
    <nav
      class="m-0 py-3 navbar navbar-expand-lg navbar-dark bg-primary"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          NEX Food Ordering
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          />
          <a
            class="nav-link"
            href="/admin"
          >
            ADMIN
          </a>
          <a
            class="nav-link"
            href="/"
          >
            Sign In
          </a>
          <a
            class="nav-link"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
    <div
      class="form-container"
    >
      <div
        class="form-box"
      >
        <div
          class="header-form"
        >
          <h4
            class="text-primary text-center"
          >
            <i
              class="fa fa-user-circle"
              style="font-size: 110px;"
            />
          </h4>
          <div
            class="image"
          />
        </div>
        <h2>
          Admin Sign In
        </h2>
        <div
          class="body-form"
        >
          <form
            class="m-3"
          >
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-envelope fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="id"
                placeholder="Admin ID"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-lock fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-secondary btn-block"
              type="submit"
            >
              Log In
            </button>
            <br />
            <br />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
